scripts/calculator.js

```diff
@@ -0,0 +1,90 @@
+"use strict";
+
+import { Display } from "./display.js";
+
+export class Calculator {
+
+    constructor(displayID, btnsID) {
+        this.display = new Display(displayID);
+        this.btns = document.querySelector(`#${btnsID}`);
+        this.abortController = new AbortController();
+
+        this.evaluator = new Worker("/scripts/workers/evaluator.js");
+
+        this.init();
+    }
+
+    // Add event handlers.
+    init() {
+        this.btns.addEventListener("click", e => this.handleClickEvent(e), { signal: this.abortController.signal });
+        document.addEventListener("keydown", e => this.handleKeyEvents(e), { signal: this.abortController.signal });
+        this.evaluator.addEventListener("message", e => this.handleResult(e), { signal: this.abortController.signal });
+    }
+
+    // Handle click events from buttons using event delegation.
+    handleClickEvent(e) {
+        const btn = e.target.closest("button.btn");
+
+        if (!btn?.value) return;
+
+        this.handleInput(btn.value);
+    }
+
+    // Handle Key Events.
+    handleKeyEvents(e) {
+        this.handleInput(e.key);
+    }
+
+    // Actual Logic to handle input from both Keyboard and Button click
+    handleInput(input) {
+        if (typeof input !== "string")
+            throw new TypeError("Input must be string");
+
+        if (!isNaN(input)) {
+            this.display.append(input);
+            return;
+        }
+
+        input = input.toLowerCase();
+
+        switch (input) {
+            case "c":
+            case "clear":
+                this.display.clear();
+                return;
+            case "backspace":
+                this.display.backspace();
+                return;
+            case "+":
+            case "-":
+            case "*":
+            case "/":
+            case ".":
```

Comment:
Can we handle the multiple decimal input issue directly with `regex`? Replacing [this](https://github.com/dhruvdhaduk-simform/scientific-calc/pull/3/commits/34b1701713dc2a41041ee8dc43de42ac51891d83#diff-ad1f93112bef47b879633d364b2363c5a1c0e6e298a22697394629b903f38301R37-R51) functionality entirely.

```suggestion
            case ".":
                if (this.display.get().match(/\.\d*$/)) return;
                this.display.append(input);
                return;
```

Reply:
Used the regex. However, I placed this validation inside `append` function of `display.js` where all validations are implemented.
https://github.com/dhruvdhaduk-simform/scientific-calc/pull/3/commits/aff247030e3c715d6f45d7fe06b4686b986c531f